### PR TITLE
cloudflared: 2022.8.2 -> 2022.11.1, add tests and additional platforms

### PR DESCRIPTION
**Pull request by piperswe**
###### Description of changes

Updated cloudflared to 2022.11.1, added tests, tweaked and updated metadata a bit to reflect cross-compilability and new features. I added myself as a maintainer - I also maintain the Homebrew package and already have notifications set up for new versions.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
  - [x] i686-w64-mingw32
    - `nix build github:piperswe/nixpkgs/cloudflared-2022.8.4#legacyPackages.x86_64-linux.cloudflared.passthru.tests.runs-through-wine`
  - [x] x86_64-w64-mingw32
    - `nix build github:piperswe/nixpkgs/cloudflared-2022.8.4#legacyPackages.x86_64-linux.cloudflared.passthru.tests.runs-through-wine64`
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by an-empty-string — APPROVED** [review]
LGTM from an x86_64-linux perspective - this builds + passes checkPhase fine, and cloudflared works with my usual helper scripts over here. I can't merge anything, but appreciate the PR :)

We are up to 2022.10.3 now, though, so it may be worth bumping the version again. (I'd submit a new PR myself for this, but don't want to have too many open ones related to this package at a time)

File: pkgs/applications/networking/cloudflared/tests.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      nativeBuildInputs = [ cloudflared ];
+    } ''
+    set -e
+    cloudflared update 2>&1 | tee output.txt
+    if ! grep "cloudflared was installed by nixpkgs" output.txt
+    then
+      echo "cloudflared's output didn't contain the package manager name"
+      exit 1
+    fi
+    mkdir $out
```
Comment (Mindavi) [inline pkgs/applications/networking/cloudflared/tests.nix:22]: touch $out is also often used for this, but presumably either is fine

File: pkgs/applications/networking/cloudflared/default.nix [hunk 2]
```diff
@@ -17,22 +17,42 @@ buildGoModule rec {
...
+      --replace "TestICMPRouterRejectNotEcho" "SkipICMPRouterRejectNotEcho"
+
+    # Workaround for: icmp_posix_test.go:28: socket: permission denied
+    substituteInPlace "ingress/icmp_posix_test.go" \
+      --replace "TestFunnelIdleTimeout" "SkipFunnelIdleTimeout"
   '';
 
   doCheck = !stdenv.isDarwin;
 
+  passthru.tests = callPackage ./tests.nix { inherit version; };
```
Comment (Mindavi) [inline pkgs/applications/networking/cloudflared/default.nix:49]: Everything looks good other than this; I am not sure if this is the right pattern, hopefully someone else can shed some light on that.

Comment (m1cr0man) [reply pkgs/applications/networking/cloudflared/default.nix:63]: First time looking into this too. It seems writing tests like this is a [documented pattern](https://ryantm.github.io/nixpkgs/contributing/coding-conventions/#ssec-inline-package-tests-writing) however you should add your own key for the test itself e.g. `passthru.tests.simple`. [home-assistant](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/home-assistant/default.nix#L361) seems to be a decent example of this.

```suggestion
  passthru.tests.simple = callPackage ./tests.nix { inherit version; };
```

**Review by m1cr0man — COMMENTED** [review]
Reviewing because I use cloudflared and I am aware of how frequently it updates. I do have a suggested change for the passthru.tests

File: pkgs/applications/networking/cloudflared/default.nix (hunk 2 above)
Comment (piperswe) [reply pkgs/applications/networking/cloudflared/default.nix]: makes sense, added your suggestion

**Pull request merged**